### ISSUE TITLE
Iconography and UI controls in Firefox iOS do not scale with Dyanmic Type

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/firefox-ios#32258]
Filing this as a placeholder, this may require a separate epic and/or an audit.

Per discussion [here](https://mozilla.slack.com/archives/C03KPTP37BN/p1771630627848389) our current a11y policy is that we should be resizing icons and UI controls along with Dynamic Type. (This also appears to match the suggested behaviors from Apple’s HIG).

In testing most of Firefox iOS, our UI generally does not appear to scale properly. This is true currently across several areas of the UI (tab tray, address bar, Settings, etc.)

We should identify which areas of the app (if not all of them) we want to scale with DT and update our UI to support this.

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/FXIOS-14974)
